### DISCUSSION TITLE
prefill chat username

### DISCUSSION THREAD

**Discussion by justine-dequidt** [owncast/owncast#3273] [Q&A] [answered]
Hello, 

I want to prefill chat username, the live will be available on my website only when user is login with her account. It's possible to prefill the username for the chat user with a url param or iframe param ? 

Thanks you for you answer, 

Cordialy,
Justine

**Comment by gabek** [comment]
If your auth proxy forwards along the `X-Forwarded-User` header that includes the username, it will be used when the user registers to chat for the first time.

https://owncast.online/docs/sslproxies/#inherit-display-name-from-reverse-proxy

**Comment by justine-dequidt** [comment]
Thank you for your answer,

I had Caddy proxy on my Owncast instance. Now I do not understand how to pass the username.
I tried passing it as a parameter: [https://my-owncast-url?X-Forwarded-User=userName], but it's not working. 

How do I need to do specific development on Owncast to retrieve the URL parameter and fill the username ? 

Do you have more information for passing username data ?

**Comment by gabek** [comment, marked as answer]
It's not through a URL parameter, it's a HTTP header. It would be used if you had an authentication proxy in front of Owncast, for example, to make them have to log in before they can access the stream. This authentication proxy would add the `X-Forwarded-User` header. I'm not sure what your auth setup is, so I can't give specifics, but whatever you're using as an authentication proxy should be able to add HTTP headers.
